tests/reg_tests/test_time_spectral_naca64A010.py

```diff
@@ -0,0 +1,176 @@
+# built-ins
+import unittest
+import os
+import copy
+import numpy
+
+# MACH testing class
+from adflow import ADFLOW
+from idwarp import USMesh
+
+# import the testing utilities that live a few directories up
+import reg_test_utils as utils
+
+from reg_default_options import adflowDefOpts
+
+from reg_aeroproblems import ap_naca64A010_time_spectral
+import reg_test_classes
+
+
+baseDir = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestSolve(reg_test_classes.RegTest):
+    """
+    Tests for time spectral case for an airfoil.
+    """
+
+    N_PROCS = 2
+    ref_file = "solve_euler_time_spectral_naca64A010.json"
+
+    def setUp(self):
+
+        # Introduce a transfer class for displacement transfer from struct to aero.
+        class Transfer:
```

Comment:
In practice, would each user have to create their own Transfer class?

Reply:
The user is responsible to provide the surface grid. Here we generate the surface grid from the structural displacement. That is why we need a transfer class.